assemblyline_core/updater/helper.py

```diff
@@ -130,6 +131,19 @@ def _get_proprietary_registry_tags(self, server, image_name, auth, verify, proxi
     'harbor': HarborRegistry()
 }
 
+def get_registry_config(docker_config: DockerConfig, system_config: SystemConfig) -> Dict[str, str]:
+    server = docker_config.image.split("/", 1)[0]
+
+    # Prioritize authentication given as a system configuration
+    for registry in system_config.services.registries:
+        if server.startswith(registry['name']):
+            # Return authentication credentials and the type of registry
+            return dict(username=registry['username'], password=registry['password'], type=registry['type'])
```

Comment:
Shouldnt it used .get because we don't know which values will be present?

Reply:
For a ServiceRegistry, allow those fields are required to be set:
https://cybercentrecanada.github.io/assemblyline4_docs/odm/models/config/#serviceregistry